gapis/api/vulkan/state_rebuilder.go

```diff
@@ -681,18 +683,21 @@ func (sb *stateBuilder) createSurface(s SurfaceObjectʳ) {
 
 func (sb *stateBuilder) createDevice(d DeviceObjectʳ) {
 	enabledLayers := []Charᶜᵖ{}
-	for _, layer := range d.EnabledLayers().All() {
+	for _, k := range d.EnabledLayers().Keys() {
+		layer := d.EnabledLayers().Get(k)
 		enabledLayers = append(enabledLayers, NewCharᶜᵖ(sb.MustAllocReadData(layer).Ptr()))
 	}
 	enabledExtensions := []Charᶜᵖ{}
-	for _, ext := range d.EnabledExtensions().All() {
+	for _, k := range d.EnabledExtensions().Keys() {
+		ext := d.EnabledExtensions().Get(k)
 		enabledExtensions = append(enabledExtensions, NewCharᶜᵖ(sb.MustAllocReadData(ext).Ptr()))
 	}
 
 	queueCreate := map[uint32]VkDeviceQueueCreateInfo{}
```

Comment:
You're still iterating over this map on :724 below..

Reply:
Fixed.